client/components/select-dropdown/test/index.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+import { assert } from 'chai';
+import { shallow, mount } from 'enzyme';
+import sinon from 'sinon';
```

Comment:
> Yeah, I've noticed it during my exploration of your test base; it is essentially a mocha-aware wrapper around sinon.sandbox.create.
> Are you suggesting it's better to use it even in a such simple scenario?

`useSandbox` takes care of tear down for you. It just simplifies a lot when you are using stubs and mocks extensively. It's fine to restore a stub inside the single test as it's done in this PR :) I don't have a preferred method here. In some cases it's also a good idea to use `sinon.test` which mixes both approaches.

> Also, I just noticed that for the assertions on spy/stub you never use sinon.assert.
> I usually prefer sinon.assert cause it's more descriptive than chai, qunit (whatever...) when an assertion fails.

Chai wraps `sinon.assert` so it should have similar handling. I'm not sure I have never used `sinon.assert` directly. We have developers with different levels of experience with unit testing. It definitely helps to have one style of writing assertions and one place to look for examples and documentation. That's why I think it's nice that we can use `sinon-chai`. On the other hand I don't see anything bad with using `sinon.assert` directly :)